### PR TITLE
Update Component Mapping with links and latest status

### LINKED ISSUE #29345: [Bug]: Component Mapping page is out of date
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
n/a
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-component-mapping--page

### Bug Description

The [Component Mapping](https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-component-mapping--page) on our react doc site is a bit out of date, there are a few components we have released to stable that are not listed. 

Also, more and more folks are referencing this page, so we have an opportunity to improve it with links out to the right components, guides, or the epic if a component is in progress. 

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by JustSlone**
## Previous Behavior

[Component Mapping](https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-component-mapping--page) was out of date since we've released a few components since it was updated. 

## New Behavior

Component Mapping page is updated and now links out to the component documentation pages. 
It also links to in progress components on our public roadmap. 

## Related Issue(s)

- Fixes #29345

### CODE REVIEW

**Comment by JustSlone** [comment]
Example result:
image

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 1]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| SwatchColorPicker      | _[In Progess](https://github.com/orgs/microsoft/projects/786/views/1?pane=issue&itemId=33879887)_                                            |
+| TagPicker              |                                                                                                                                              |
+| TeachingBubble         | _[In Progress](https://github.com/orgs/microsoft/projects/786/views/1?pane=issue&itemId=24403213)_                                           |
+| Text                   | [Text](/docs/components-text--default)                                                                                                       |
+| TextField              | [Input](/docs/components-input--default)                                                                                                     |
+| ThemeProvider          | [FluentProvider](/docs/components-fluentprovider--default)                                                                                   |
+| ToggleButton           | [ToggleButton](/docs/components-button-togglebutton--default)                                                                                |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:80]: Toggle -> Switch missing

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 2]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| Stack                  | [Migration Guide](/docs/concepts-migration-from-v8-components-flex-stack--page), [StackShim](/docs/migration-shims-v8-stackshim--playground) |
+| SwatchColorPicker      | _[In Progess](https://github.com/orgs/microsoft/projects/786/views/1?pane=issue&itemId=33879887)_                                            |
+| TagPicker              |                                                                                                                                              |
+| TeachingBubble         | _[In Progress](https://github.com/orgs/microsoft/projects/786/views/1?pane=issue&itemId=24403213)_                                           |
+| Text                   | [Text](/docs/components-text--default)                                                                                                       |
+| TextField              | [Input](/docs/components-input--default)                                                                                                     |
+| ThemeProvider          | [FluentProvider](/docs/components-fluentprovider--default)                                                                                   |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:79]: TimePicker missing

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 3]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| DetailsList            | [~DataGrid](/docs/components-datagrid--default)                                                                                              |
+| Dialog                 | [Dialog](/docs/components-dialog--default)                                                                                                   |
+| DocumentCard           | [~Card](/docs/components-card-card--default)                                                                                                 |
+| Dropdown               | [~Dropdown](/docs/components-dropdown--default)                                                                                              |
+| Facepile               | [~AvatarGroup](/docs/components-avatargroup--default)                                                                                        |
+| GroupedList            | [~Tree](/docs/components-tree--default)                                                                                                      |
+| Icon                   | [(icons in react-icons)](/docs/concepts-developer-icons-icons--page)                                                                         |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:45]: HoverCard missing

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 4]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+|                        | [Accordion](/docs/components-accordion--default)                                                                                             |
+|                        | [Badge](/docs/components-badge-badge--default)                                                                                               |
+|                        | [CounterBadge](/docs/components-badge-counter-badge--default)                                                                                |
+|                        | [PresenceBadge](/docs/components-badge-presencebadge--default)                                                                               |
+| ActionButton           | [Button](/docs/components-button-button--default)                                                                                            |
+| ActivityItem           |                                                                                                                                              |
+| BasicList              | _[In Progress](https://github.com/orgs/microsoft/projects/786/views/1?pane=issue&itemId=24404181)_                                           |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:22]: The docs call it basic list, but the v8 component name is List

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 5]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| Breadcrumb             | _[In Progress](https://github.com/orgs/microsoft/projects/786/views/1?pane=issue&itemId=24402993)_                                           |
+| CommandBarButton       |                                                                                                                                              |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:24]: CommandBar -> Toolbar missing

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx (hunk 1 above)
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:80]: Tooltip missing

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx (hunk 4 above)
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:22]: Maybe Announced is missing or maybe we don't want to list that one.

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 6]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| DatePicker             | [DatePickerCompat](/docs/compat-components-datepicker--default)                                                                              |
+| DetailsList            | [~DataGrid](/docs/components-datagrid--default)                                                                                              |
+| Dialog                 | [Dialog](/docs/components-dialog--default)                                                                                                   |
+| DocumentCard           | [~Card](/docs/components-card-card--default)                                                                                                 |
+| Dropdown               | [~Dropdown](/docs/components-dropdown--default)                                                                                              |
+| Facepile               | [~AvatarGroup](/docs/components-avatargroup--default)                                                                                        |
+| GroupedList            | [~Tree](/docs/components-tree--default)                                                                                                      |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:44]: FocusTrapZone, FocusZone -> Tabster ?

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 7]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| Icon                   | [(icons in react-icons)](/docs/concepts-developer-icons-icons--page)                                                                         |
+| IconButton             | [Button](/docs/components-button-button--default)                                                                                            |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:46]: Icon -> react-icons package ?

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 8]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| Image                  | [Image](/docs/components-image--default)                                                                                                     |
+| Label                  | [Label](/docs/components-label--default)                                                                                                     |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:48]: Keytips missing or maybe we don't want to mention that one either :)

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx [hunk 9]
```diff
@@ -11,70 +11,70 @@ Here is the mapping of v8 components to their v9 component replacement or equiva
...
+| Panel                  | _[In Progess](https://github.com/orgs/microsoft/projects/786/views/1?pane=issue&itemId=24402572)_                                            |
+| PrimaryButton          | [Button](/docs/components-button-button--default)                                                                                            |
+| PeoplePicker           |                                                                                                                                              |
+| Persona                | [Avatar](/docs/components-avatar--default)                                                                                                   |
+| Pickers                |                                                                                                                                              |
+| Pivot, PivotItem       | [TabList, Tab](/docs/components-tablist--default)                                                                                            |
+| ProgressIndicator      | [ProgressBar](/docs/components-progressbar--default)                                                                                         |
```
Comment (GeoffCoxMSFT) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:62]: Popup Missing

File: apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx (hunk 8 above)
Comment (JustSlone) [reply apps/public-docsite-v9/src/Concepts/Migration/FromV8/ComponentMapping.stories.mdx:48]: Who the heck made this list, so many things missing 👀?! /s

**Comment by JustSlone** [comment]
@GeoffCoxMSFT All comments resolved! Thanks for the help

**Review by GeoffCoxMSFT — APPROVED** [review]

**Pull request merged**